sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -181,14 +196,32 @@ def __iter__(self):
         windowed_key_values = trigger_driver.process_entire_key
       coder_impl = self._post_grouped_coder.get_impl()
       key_coder_impl = self._key_coder.get_impl()
-      for encoded_key, windowed_values in self._table.items():
+      n = min(n, len(self._table))
+      output_stream_list = []
+      for _ in range(n):
+        output_stream_list.append(create_OutputStream())
+      for idx, (encoded_key, windowed_values) in enumerate(self._table.items()):
         key = key_coder_impl.decode(encoded_key)
         for wkvs in windowed_key_values(key, windowed_values):
-          coder_impl.encode_to_stream(wkvs, output_stream, True)
-      self._grouped_output = [output_stream.get()]
+          coder_impl.encode_to_stream(wkvs, output_stream_list[idx % n], True)
+      for output_stream in output_stream_list:
+        self._grouped_output.append([output_stream.get()])
       self._table = None
     return iter(self._grouped_output)
 
+  def __iter__(self):
+    """ Since partition() returns a list of list, added this __iter__ to return
+    a list to simplify code when we need to iterate through ALL elements of
+    _GroupingBuffer.
+    """
+    if len(self._grouped_output) == 0:
+      self.partition(1)
```

Comment:
If calling a method just for its side effects, it probably makes sense to have a comment.